src/Storages/StorageKeeperMap.cpp

```diff
@@ -342,45 +372,29 @@ StorageKeeperMap::StorageKeeperMap(
             }
         }
 
-        std::string stored_metadata_string;
-        auto exists = client->tryGet(metadata_path, stored_metadata_string);
-
-        if (exists)
+        Coordination::Requests create_requests
         {
-            // this requires same name for columns
-            // maybe we can do a smarter comparison for columns and primary key expression
-            if (stored_metadata_string != metadata_string)
-                throw Exception(
-                    ErrorCodes::BAD_ARGUMENTS,
-                    "Path {} is already used but the stored table definition doesn't match. Stored metadata: {}",
-                    root_path,
-                    stored_metadata_string);
-        }
-        else
+            zkutil::makeCreateRequest(metadata_path, metadata_string, zkutil::CreateMode::Persistent),
+            zkutil::makeCreateRequest(data_path, metadata_string, zkutil::CreateMode::Persistent),
+            zkutil::makeCreateRequest(tables_path, "", zkutil::CreateMode::Persistent),
+            zkutil::makeCreateRequest(table_path, "", zkutil::CreateMode::Persistent),
+        };
+
+        Coordination::Responses create_responses;
+        auto code = client->tryMulti(create_requests, create_responses);
+        if (code == Coordination::Error::ZNODEEXISTS)
         {
-            auto code = client->tryCreate(metadata_path, metadata_string, zkutil::CreateMode::Persistent);
-            if (code == Coordination::Error::ZNODEEXISTS)
-                continue;
-            else if (code != Coordination::Error::ZOK)
-                throw Coordination::Exception(code, metadata_path);
+            LOG_WARNING(log, "It looks like a table on path {} was created by another server at the same moment, will retry", root_path);
```

Comment:
It's rather info than warning (for example, it may normally happen on `CREATE ON CLUSTER`)